### ISSUE TITLE
MappingException when enum toString() is overridden

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#49]
```
What steps will reproduce the problem?
1.Create a enum and override toString method. 
2.Create a object with the enum field.
2.Map this object to another object with corresponding enum field.

What is the expected output? What do you see instead?
Filled object is  expected. I see MappingException instead.

What version of the product are you using? On what operating system?
Orika 1.2.0. Windows 7 and Ubuntu 12.0.4

Please provide any additional information below.
see attached test and log.
```

Original issue reported on code.google.com by `dkhomya...@gmail.com` on 11 Sep 2012 at 12:13

Attachments:
- [enumTest.log](https://storage.googleapis.com/google-code-attachments/orika/issue-49/comment-0/enumTest.log)
- [EnumTest.java](https://storage.googleapis.com/google-code-attachments/orika/issue-49/comment-0/EnumTest.java)

**Comment by GoogleCodeExporter** [comment]
```
Ok, in this case, I guess we have to just to skip it, and let the user specify 
a custom converter instead of automatic mapping?
```

Original comment by `elaat...@gmail.com` on 11 Sep 2012 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
Unfortunally, custom converter isn't suitable for me. I need to develop api 
with automatic entity-dto mapping. I suppose,  cause of the problem is Orika 
use Enum.toString()(not final) instead of Enum.name()(final).  That seems easy 
for fix.
```

Original comment by `dkhomya...@gmail.com` on 11 Sep 2012 at 1:25

**Comment by GoogleCodeExporter** [comment]
```
Of course, I was mission the point.
I was thinking that when you have different literal names (between two 
Enumerations) .
This can will be shipped wihin next release (in the next few days)
```

Original comment by `elaat...@gmail.com` on 11 Sep 2012 at 2:17
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Of course, I was mission the point.
I was thinking that when you have different literal names (between two 
Enumerations) .
This can will be shipped wihin next release (in the next few days)
```

Original comment by `elaat...@gmail.com` on 11 Sep 2012 at 2:17

**Comment by GoogleCodeExporter** [comment]
Original comment by `elaat...@gmail.com` on 11 Sep 2012 at 7:11
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
With your permission I will include your test to Orika,

Thanks in advance 
```

Original comment by `elaat...@gmail.com` on 11 Sep 2012 at 7:52
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for responsiveness!
```

Original comment by `dkhomya...@gmail.com` on 12 Sep 2012 at 6:50

**Comment by GoogleCodeExporter** [comment]
```
A fix has been shipped within 1.2.1 release
this release ship some performance improvements!
```

Original comment by `elaat...@gmail.com` on 15 Sep 2012 at 3:55
